packages/offix-datastore/src/DataStore.ts

```diff
@@ -35,34 +31,34 @@ export class DataStore {
   private dbName: string;
   private schemaVersion: number;
   private models: Model<unknown>[];
-  private storage?: LocalStorage;
-  private url: string;
-  private wsUrl?: string | undefined;
+  private indexedDB: IndexedDBStorageAdapter;
+  private storage: LocalStorage;
+  private clientConfig: any;
+  private metadataName = "metadata";
 
   constructor(config: DataStoreConfig) {
     this.dbName = config.dbName;
-    this.schemaVersion = config.schemaVersion || 1; // return 1 is schemaVersion is undefined or 0
-    this.url = config.url;
-    this.wsUrl = config.wsUrl || undefined;
+    this.schemaVersion = config.schemaVersion || 1; // return 1 if schemaVersion is undefined or 0
+    this.clientConfig = config.clientConfig;
     this.models = [];
+    this.indexedDB = new IndexedDBStorageAdapter();
+    this.storage = new LocalStorage(this.indexedDB);
+    this.indexedDB.addStore({ name: this.metadataName });
```

Comment:
@Eunovo yeah, this makes sense to have. Allows us to be slightly more flexible. Good job